lib/rules/color-hex-length/index.js

```diff
@@ -117,12 +105,23 @@ function longer(hex) {
 	return hexVariant;
 }
 
-function replaceHex(input, searchString, replaceString, startIndex) {
-	const offset = startIndex + 1;
-	const stringStart = input.slice(0, offset);
-	const stringEnd = input.slice(offset + searchString.length);
+function isIgnoredFunction({ type, value }) {
+	return type === 'function' && IGNORED_FUNCTIONS.has(value.toLowerCase());
+}
+
+function isHexColor({ type, value }) {
+	return type === 'word' && HEX.test(value);
+}
+
+function getValue(decl) {
+	return decl.raws.value ? decl.raws.value.raw : decl.value;
+}
+
+function setValue(decl, value) {
+	if (decl.raws.value) decl.raws.value.raw = value;
+	else decl.value = value;
 
-	return stringStart + replaceString + stringEnd;
+	return decl;
```

Comment:
```suggestion
```

We can remove this redundant return as per the other pull request.

At some point later down the line, we should refactor to move at least the `getValue` and `setValue` functions out to `/utils`. For now, let's duplicate them as you are doing, though.